Add basic documentation (#21)
qutip/qutip-jax: doc/source/basics.rst

.. _qtjax_intro:

************
Jax in Qutip
************

.. _basic_usage:

Basic Usages
============

> Once imported, jax array will be available to contain ``qutip.Qobj``'s data and diffrax ODE will be available for solver (``sesolve``, ``mcsolve``, etc.).
> None of the function in the module are expected to be used directly, they will be used by qutip when the options to use this backend is used.

Reviewer: ```suggestion
In orther to enable qutip-jax, it is just necessary to import the module. Once imported, ``qutip.Qobj``'s data can be represented as a JAX array. Furthermore, diffrax ODE will be available as an option for qutip's solvers (``sesolve``, ``mcsolve``, etc.).
None of the functions in the module are expected to be used directly. Instead, they will be used by qutip, allowing the user to interact only with the already familiar QuTiP interface.
```

> There are many ways to tell Qutip to use the jax backend.

Reviewer: ```suggestion
There are many ways to create a QuTiP ``Qobj`` backed by JAX's array class.
```

> - Pass Jax array to the Qobj constructor.

Reviewer: ```suggestion
- Passing JAX array to the ``Qobj`` constructor.
```

.. code-block::

    import qutip
    import qutip_jax
    import jax.numpy as jnp

>     jax_eye_qobj = qutip.Qobj(jnp.eye(3))

Reviewer: Perhaps in these codeblocks we can add something at the end (`jax_eye_qobj.data` for example) to make it easier to whoever is executing this code what to expect or how to assess it jax is being used.

Author: Done, was it what you had in mind?

Reviewer: Yes, looks great now.

> - Manually convert a ``Qobj`` or ``QobjEvo``.

Reviewer: ```suggestion
- Converting a ``Qobj`` or ``QobjEvo`` using the ``to`` method.
```

.. code-block::

    import qutip
    import qutip_jax

    jax_eye_qobj = qutip.qeye(3).to("jax")

> - Use qutip state and operator function with ``dtype="jax"``.

Reviewer: ```suggestion
- Using QuTiP's native constructors' ``dtype`` parameter.
```

.. code-block::

    import qutip
    import qutip_jax

    jax_eye_qobj = qutip.qeye(3, dtype="jax")

> - Use qutip settings to set jax as the default backend.

Reviewer: ```suggestion
- Changing QuTiP's default settings to set JAX as the default backend.
```

.. code-block::

    import qutip
    import qutip_jax
    qutip.settings.core["default_dtype"] = "jax"

    jax_eye_qobj = qutip.qeye(3)

> - Use qutip settings in a context.

Reviewer: ```suggestion
- Changing QuTiP's default settings within a context.
```

.. _qtjax_intro:

************

> Jax in Qutip

Reviewer: ```suggestion
Jax in QuTiP
```

************

.. _basic_usage:

> Basic Usages
> ============

Reviewer: ```suggestion
Basic usage
===========
```
